translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -9572,9 +9747,83 @@ Value *SPIRVToLLVM::transGLSLExtInst(SPIRVExtInst *ExtInst,
     return getBuilder()->CreateBitCast(
         Args[0], VectorType::get(getBuilder()->getInt32Ty(), 2));
 
-  default:
-    // Other instructions are handled the old way, by generating a call.
+  case GLSLstd450Length: {
+    // Get length of vector.
+    if (!isa<VectorType>(Args[0]->getType())) {
+      return getBuilder()->CreateUnaryIntrinsic(Intrinsic::fabs, Args[0]);
+    }
+    Value *Dot = getBuilder()->CreateDotProduct(Args[0], Args[0]);
+    return getBuilder()->CreateUnaryIntrinsic(Intrinsic::sqrt, Dot);
+  }
+
+  case GLSLstd450Distance: {
+    // Get distance between two points.
+    Value *Diff = getBuilder()->CreateFSub(Args[0], Args[1]);
+    if (!isa<VectorType>(Diff->getType())) {
+      return getBuilder()->CreateUnaryIntrinsic(Intrinsic::fabs, Diff);
+    }
+    Value *Dot = getBuilder()->CreateDotProduct(Diff, Diff);
+    return getBuilder()->CreateUnaryIntrinsic(Intrinsic::sqrt, Dot);
+  }
+
+  case GLSLstd450Cross:
+    // Vector cross product.
+    return getBuilder()->CreateCrossProduct(Args[0], Args[1]);
+
+  case GLSLstd450Normalize:
+    // Normalize vector to magnitude 1.
+    return getBuilder()->CreateNormalizeVector(Args[0]);
+
+  case GLSLstd450FaceForward:
+    // Face forward operation.
+    return getBuilder()->CreateFaceForward(Args[0], Args[1], Args[2]);
+
+  case GLSLstd450Reflect:
+    // Reflect operation.
+    return getBuilder()->CreateReflect(Args[0], Args[1]);
+
+  case GLSLstd450Refract:
+    // Refract operation.
+    return getBuilder()->CreateRefract(Args[0], Args[1], Args[2]);
+
+  case GLSLstd450FindILsb: {
+    // Find integer least-significant 1-bit.
+    // The spec claims that the result must be the same type as the input, but I
+    // have seen SPIR-V that does not do that.
+    Value *Result = getBuilder()->CreateBinaryIntrinsic(
+        Intrinsic::cttz, Args[0], getBuilder()->getTrue());
+    return getBuilder()->CreateZExtOrTrunc(Result,
+                                           transType(ExtInst->getType()));
+  }
+
+  case GLSLstd450FindSMsb: {
+    // Find signed integer most-significant bit.
+    // The spec claims that the result must be the same type as the input, but I
+    // have seen SPIR-V that does not do that.
```

Comment:
- Where did you find the SPIR-V binary? games or CTS? Is it possible to add it as comment?
- What's the validation result with spirv-tools?